Kickstarter-iOS/Views/Controllers/CancelPledgeViewController.swift

```diff
@@ -14,20 +33,190 @@ final class CancelPledgeViewController: UIViewController {
+
+    _ = ([
+      self.cancellationDetailsTextLabel,
+      self.cancellationReasonTextField,
+      self.cancellationReasonDisclaimerLabel,
+      self.cancelButton,
+      self.goBackButton
+    ], self.rootStackView)
+      |> ksr_addArrangedSubviewsToStackView()
+
+    _ = (self.cancellationReasonDisclaimerLabel, self.rootStackView)
+      |> ksr_setCustomSpacing(Styles.grid(10))
+
+    _ = (self.cancellationDetailsTextLabel, self.rootStackView)
+      |> ksr_setCustomSpacing(Styles.grid(6))
+
+    _ = (self.cancellationReasonTextField, self.rootStackView)
+      |> ksr_setCustomSpacing(Styles.grid(1))
+
+    self.setupConstraints()
+
+    self.view.addGestureRecognizer(
+      UITapGestureRecognizer(target: self, action: #selector(CancelPledgeViewController.dismissKeyboard))
+    )
+
+    self.goBackButton.addTarget(
+      self, action: #selector(CancelPledgeViewController.goBackButtonTapped),
+      for: .touchUpInside
+    )
+
+    self.viewModel.inputs.viewDidLoad()
+  }
+
+  override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+    super.traitCollectionDidChange(previousTraitCollection)
+
+    self.viewModel.inputs.traitCollectionDidChange()
   }
 
   // MARK: - Configuration
 
-  internal func configure(with _: Project, backing _: Backing) {}
+  internal func configure(with project: Project, backing: Backing) {
+    self.viewModel.inputs.configure(with: project, backing: backing)
+  }
 
   override func bindStyles() {
     super.bindStyles()
 
     _ = self.view
-      |> \.backgroundColor .~ UIColor.ksr_grey_400
+      |> checkoutBackgroundStyle
+
+    _ = self.scrollView
+      |> \.alwaysBounceVertical .~ true
+      |> \.showsVerticalScrollIndicator .~ false
+      |> \.contentInset %~ { _ -> UIEdgeInsets in
+        self.contentInsetsFor(traitCollection: self.traitCollection)
+      }
+
+    _ = self.rootStackView
+      |> checkoutRootStackViewStyle
+      |> \.spacing .~ Styles.grid(2)
+
+    _ = self.cancellationDetailsTextLabel
+      |> cancellationDetailsTextLabelStyle
+
+    _ = self.cancellationReasonTextField
+      |> cancellationReasonTextFieldStyle
+      |> UITextField.lens.placeholder %~ { _ in Strings.Tell_us_why_optional() }
+
+    _ = self.cancellationReasonDisclaimerLabel
+      |> cancellationDisclaimerLabelStyle
+      |> \.text %~ { _ in Strings.We_wont_share_this_with_the_creator() }
+
+    _ = self.cancelButton
+      |> redButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.Yes_cancel_it() }
+
+    _ = self.goBackButton
+      |> greyButtonStyle
+      |> UIButton.lens.title(for: .normal) %~ { _ in Strings.No_go_back() }
   }
 
   override func bindViewModel() {
     super.bindViewModel()
+
+    self.viewModel.outputs.cancellationDetailsTextLabelValue
+      .observeForUI()
+      .observeValues { [weak self] amount, projectName in
+        self?.setCancellationDetailsAttributedText(with: amount, projectName: projectName)
```

Comment:
Shouldn't this attributed text be returned from the `vm` so that we can simply use reactive binding?